### ISSUE TITLE
Can't update small in 1.10

### ISSUE THREAD

**Issue by Gerold103** [tarantool/tarantool#5060]
Try to update small to master in 1.10, cmake, and get the error:
```
CMake Error at test/unit/CMakeLists.txt:21 (add_executable):
  add_executable cannot create target "rlist.test" because another target
  with the same name already exists.  The existing target is an executable
  created in source directory
  "/Users/gerold/Work/Repositories/tarantool/src/lib/small/test".  See
  documentation for policy CMP0002 for more details.

CMake Error at test/unit/CMakeLists.txt:22 (target_link_libraries):
  Attempt to add link library "unit" to target "rlist.test" which is not
  built in this directory.

  This is allowed only when policy CMP0079 is set to NEW.
```
Seems like something was changed in small's build scripts, but it wasn't updated in 1.10.

After this is fixed, to make compilation work need to cherry-pick to 1.10 this https://github.com/tarantool/tarantool/commit/ffec9e40d8695bffe76109b721b1050db84b384e and this https://github.com/tarantool/tarantool/commit/635b5ad9f84120fa023b80d503f075db3c2be4c9.

Small master contains a bug fix for vinyl, so it is important to find a way to update it in 1.10.

**Comment by cyrillos** [comment]
Sent out and pushed into branch `gorcunov/gh-5060-small-110`

**Comment by cyrillos** [comment]
Merged into 1.10
